dashboard/test/lib/user_multi_auth_helper_test.rb

```diff
@@ -186,42 +208,40 @@ def assert_convert_oauth_user(user)
     initial_oauth_token = user.oauth_token
     initial_oauth_token_expiration = user.oauth_token_expiration
 
+    is_email_trusted = %w(facebook google_oauth2 windowslive).include? provider
+
+    # Assert email remains empty for students
+    expected_email = user.student? ? :empty : initial_email
+
+    # Before migration hashed_email can be nil for untrusted provide
+    # But it's not-nullable in AuthenticationOption
+    expected_hashed_email = initial_hashed_email.nil? ? '' : initial_hashed_email
+
     assert_user user,
       provider: provider,
       email: user.student? ? :empty : :not_empty,
-      hashed_email: :not_empty,
+      hashed_email: is_email_trusted ? :not_empty : initial_hashed_email,
       uid: :not_nil,
       oauth_token: :not_nil,
       oauth_token_expiration: :not_nil
 
     migrate user
 
     assert_user user,
-      email: user.student? ? :empty : initial_email,
-      hashed_email: initial_hashed_email,
+      email: expected_email,
+      hashed_email: expected_hashed_email,
       primary_authentication_option: {
         credential_type: provider,
         authentication_id: initial_authentication_id,
-        email: user.student? ? :empty : initial_email,
-        hashed_email: initial_hashed_email,
+        email: expected_email,
+        hashed_email: expected_hashed_email,
         data: {
           oauth_token: initial_oauth_token,
           oauth_token_expiration: initial_oauth_token_expiration
         }
       }
   end
 
-  #
-  # Untrusted email from Oauth:
-  #
-
-  # TODO: Clever Oauth user
-  # TODO: Clever Oauth user that also has a password (due to takeover)
-  # TODO: Powerschool Oauth
-  # TODO: Powerschool Oauth user that also has a password (due to takeover)
```

Comment:
As I thought through the migrations, I realized having a password (or not) is totally independent of the migration logic, so I didn't implement separate tests for these cases.  Let me know if you disagree.